azure_jumpstart_ag/scripts/preprovision.ps1

```diff
@@ -0,0 +1,162 @@
+            }
+        }
+
+    $usableLocations
+}
+#endregion Functions
+
+$location = $env:AZURE_LOCATION
+$minCores = 32
+$skuFriendlyNames = "Standard DSv5 Family vCPUs|Total Regional vCPUs"
+
+Write-Host "`nChecking for available capacity in $location region..."
+
+$available = Get-AzAvailableCores -location $location -skuFriendlyNames $skuFriendlyNames -minCores $minCores
+
+If ($available.usableLocation -contains $false) {
+    Write-Host "`n`u{274C} There is not enough VM capacity in the $location region to deploy the Jumpstart environment." -ForegroundColor Red
+    
+    Write-Host "`nChecking other regions in the same geography with enough capacity ($minCores cores)...`n"
+
+    $locations = Get-AzAvailableLocations -location $location -skuFriendlyNames $skuFriendlyNames -minCores $minCores | 
+        Format-Table Location, DisplayName, TotalCores, AvailableCores, UsableLocation -AutoSize | Out-String
+    
+    Write-Host $locations
+
+    Write-Host "Please run ``azd env --new`` to create a new environment and select the new location.`n"
+
+    $message = "Not enough capacity in $location region."
+    Throw $message
+
+} else {
+    Write-Host "`n`u{2705} There is enough VM capacity in the $location region to deploy the Jumpstart environment.`n"
+}
+
+
+########################################################################
+# Get Windows Admin Username and Password
+########################################################################
+$JS_WINDOWS_ADMIN_USERNAME = 'arcdemo'
+if ($promptOutput = Read-Host "Enter the Windows Admin Username [$JS_WINDOWS_ADMIN_USERNAME]") { $JS_WINDOWS_ADMIN_USERNAME = $promptOutput }
+
+# set the env variable
+azd env set JS_WINDOWS_ADMIN_USERNAME $JS_WINDOWS_ADMIN_USERNAME
+
+# The user will be prompted for this by azd so we can maintain the security of the password.
+# $JS_WINDOWS_ADMIN_PASSWORD = Read-Host "Enter the Windows Admin Password (hint: ArcPassword123!! - 12 character minimum)" -AsSecureString
+
+# azd env set JS_WINDOWS_ADMIN_PASSWORD $JS_WINDOWS_ADMIN_PASSWORD
+
+
+########################################################################
+# Create SSH RSA Public Key
+########################################################################
+Write-Host "Creating SSH RSA Public Key..."
+$file = "js_rsa"
+remove-item $file, "$file.pub" -Force -ea 0 
+
+# Generate the SSH key pair
+ssh-keygen -q -t rsa -b 4096 -f $file -N '""' 
+
+# Get the public key
+$JS_SSH_RSA_PUBLIC_KEY = get-content "$file.pub"
+
+# Escape the backslashes 
+$JS_SSH_RSA_PUBLIC_KEY = $JS_SSH_RSA_PUBLIC_KEY.Replace("\", "\\")
+
+# set the env variable
+azd env set JS_SSH_RSA_PUBLIC_KEY $JS_SSH_RSA_PUBLIC_KEY
+
+
+########################################################################
+# Create Azure Service Principal
+########################################################################
+# TODO - consider moving SPN creation to Bicep
```

Comment:
Let's make sure to get rid of these TODO's before we will go live, unless you think it's important to keep

Reply:
It's not required to go live - but would be good to move into Issues for tracking/discussion/triage.

Reply:
Created an issue for it and removed the comment.